src/Sylius/Bundle/ThemeBundle/Context/ThemeContext.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace Sylius\Bundle\ThemeBundle\Context;
+
+use Sylius\Bundle\ThemeBundle\Model\ThemeInterface;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+class ThemeContext implements ThemeContextInterface
+{
+    /**
+     * @var ThemeInterface[]
+     */
+    private $themes = [];
+
+    /**
+     * Keys are logical names of themes, values are priorities.
+     *
+     * @var array
+     */
+    private $themesPriorities = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getThemes()
+    {
+        return $this->themes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTheme($logicalName)
+    {
+        foreach ($this->themes as $theme) {
+            if ($logicalName === $theme->getLogicalName()) {
+                return $theme;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getThemePriority($logicalName)
+    {
+        return isset($this->themesPriorities[$logicalName]) ? $this->themesPriorities[$logicalName] : null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getThemesPriorities()
+    {
+        return $this->themesPriorities;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getThemesSortedByPriorityInAscendingOrder()
+    {
+        return array_reverse($this->getThemesSortedByPriorityInDescendingOrder());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getThemesSortedByPriorityInDescendingOrder()
+    {
+        $themes = [];
+
+        $themesPriority = $this->themesPriorities;
+        arsort($themesPriority);
+
+        foreach ($themesPriority as $logicalName => $priority) {
+            $themes[] = $this->getTheme($logicalName);
+        }
+
+        return $themes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addTheme(ThemeInterface $theme, $priority = 0)
+    {
+        foreach ($this->themes as $existingTheme) {
+            if ($existingTheme->equals($theme)) {
+                return;
+            }
+        }
```

Comment:
I would identify themes only by logical name. Themes with same path sounds incorrect, but only one theme can be active at a time, so it will not break anything.

```php
if (isset($this->themes[$theme->getLogicalName()])) {
    return;
}

$this->themes[$theme->getLogicalName()] = $theme;
$this->themesPriorities[$theme->getLogicalName()] = $priority;
```